Library/OcFlexArrayLib/FlexString.c

```diff
@@ -0,0 +1,67 @@
+/** @file
+  Flexible string tokens.
+
+  Copyright (c) 2022, Mike Beaton, PMheart. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-3-Clause
+**/
+
+#include <Base.h>
+#include <Library/BaseLib.h>
+#include <Library/OcDebugLogLib.h>
+#include <Library/OcFlexArrayLib.h>
+
+OC_FLEX_ARRAY *
+OcStringSplit (
+  IN OUT  VOID           *String,
+  IN      CONST CHAR16   Delim,
+  IN      CONST BOOLEAN  IsUnicode
+  )
+{
+  OC_FLEX_ARRAY  *Tokens;
+  VOID           **Pointer;
+  VOID           *NextToken;
+  CHAR16         Ch;
+
+  ASSERT (String != NULL);
+
+  if (IsUnicode ? ((CHAR16 *) String)[0] == CHAR_NULL : ((CHAR8 *) String)[0] == '\0') {
+    return NULL;
+  }
+
+  Tokens = OcFlexArrayInit (sizeof (String), OcFlexArrayFreePointerItem);
```

Comment:
No, if you want to free with OcFlexArrayFreePointerItem you need to modify the splitter to actualy allocate space for each token as it finds it. You may prefer to do that, instead of the current destructive parsing.

Reply:
https://github.com/acidanthera/OpenCorePkg/pull/322/commits/7d9c977f4793ad0385db91b1369252bb5fd9b8f4